drivers/clock_control/clock_stm32_ll_common.c

```diff
@@ -704,7 +708,11 @@ int stm32_clock_control_init(const struct device *dev)
 
 	/* If freq increases, set flash latency before any clock setting */
 	if (old_flash_freq < CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC) {
+#if defined(CONFIG_SOC_SERIES_STM32C0X)
+		UTILS_SetFlashLatency(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
+#else
 		LL_SetFlashLatency(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
+#endif /* CONFIG_SOC_SERIES_STM32C0X */
```

Comment:
This is clearly a bug in LL API.
Please add a new commit to the HAL PR which add `LL_SetFlashLatency` as a public version of `UTILS_SetFlashLatency`.
When doing so, please add reference to the patch in the `stm32cube/stm32c0xx/README` using internal bug reference: 146704.

Reply:
Fixed